### PR TITLE
GH-4092: Add support for Environment.UserHomeDirectory

### LINKED ISSUE #4092: Add support for getting the user's home directory in the Cake Environment
The Cake Environment currently supports getting the `WorkingDirectory` as well as the `ApplicationRoot`, but doesn't support getting the user's home directory.

Proposed API:

```csharp
public interface ICakeEnvironment
{
    /// <summary>
    /// Gets the user's home directory.
    /// </summary>
    /// <value>The user's home directory.</value>
    DirectoryPath UserHomeDirectory { get; }

    // ...
}
```

### PR DESCRIPTION
**Pull request by augustoproiete**
Closes #4092

### CODE REVIEW

File: src/Cake.Core/ICakeEnvironment.cs [hunk 1]
```diff
@@ -18,6 +18,12 @@ public interface ICakeEnvironment
         /// <value>The working directory.</value>
         DirectoryPath WorkingDirectory { get; set; }
 
+        /// <summary>
+        /// Gets the user's home directory.
+        /// </summary>
+        /// <value>The user's home directory.</value>
+        DirectoryPath UserHomeDirectory { get; }
```
[the remark this answers is not in the document]
Comment (augustoproiete) [reply src/Cake.Core/ICakeEnvironment.cs:25]: @devlead It's an ABI breaking change indeed. I think moving the impl to the interface is a good idea to get this out before next major 🤔 . I'll take a stab when I get a chance

[the remark this answers is not in the document]
Comment (augustoproiete) [reply src/Cake.Core/ICakeEnvironment.cs]: @devlead Added default implementation to the interface.

**Review by patriksvensson — APPROVED** [review]
LGTM!

**Review by devlead — APPROVED** [review]
LGMT

**Pull request merged**